neural_network/activation_functions/mish.py

```diff
@@ -0,0 +1,39 @@
+"""
+Mish Activation Function
+
+Use Case: Improved version of the ReLU activation function used in Computer Vision.
+For more detailed information, you can refer to the following link:
+https://en.wikipedia.org/wiki/Rectifier_(neural_networks)#Mish
+"""
+
+import numpy as np
+
+
+def mish(vector: np.ndarray) -> np.ndarray:
+    """
+        Implements the GELU activation function.
+
+        Parameters:
+            vector (np.ndarray): The input array for Mish activation.
+
+        Returns:
+            np.ndarray: The input array after applying the Mish activation.
+
+        Formula:
+            f(x) = x * np.tanh(np.softplus(x)) = x * np.tanh(np.log(1 + np.exp(x)))
```

Comment:
```suggestion
            f(x) = x * tanh(softplus(x)) = x * tanh(ln(1 + e^x))
```
Let's have the documentation's formula be more human-readable